tpkg/util.go

```diff
@@ -255,253 +268,124 @@ func RandMilestone(parents iotago.MessageIDs) (*iotago.Milestone, []byte) {
 			copy(b[:], inclusionMerkleProof)
 			return b
 		}(),
-		PublicKeys: func() [][iotago.MilestonePublicKeyLength]byte {
-			msPubKeys := make([][iotago.MilestonePublicKeyLength]byte, sigsCount)
+		Metadata: RandBytes(10),
+		Signatures: func() iotago.Signatures {
+			msSigs := make(iotago.Signatures, sigsCount)
 			for i := 0; i < sigsCount; i++ {
-				msPubKeys[i] = Rand32ByteArray()
-				// ensure lexical ordering
-				msPubKeys[i][0] = byte(i)
-			}
-			return msPubKeys
-		}(),
-		Signatures: func() [][iotago.MilestoneSignatureLength]byte {
-			msSigs := make([][iotago.MilestoneSignatureLength]byte, sigsCount)
-			for i := 0; i < sigsCount; i++ {
-				msSigs[i] = RandMilestoneSig()
+				msSigs[i] = RandEd25519Signature()
 			}
+			sort.Slice(msSigs, func(i, j int) bool {
+				return bytes.Compare(msSigs[i].(*iotago.Ed25519Signature).PublicKey[:], msSigs[j].(*iotago.Ed25519Signature).PublicKey[:]) == -1
+			})
 			return msSigs
 		}(),
 	}
 
-	var b bytes.Buffer
-	Must(binary.Write(&b, binary.LittleEndian, iotago.MilestonePayloadTypeID))
-	Must(binary.Write(&b, binary.LittleEndian, msPayload.Index))
-	Must(binary.Write(&b, binary.LittleEndian, msPayload.Timestamp))
-	Must(binary.Write(&b, binary.LittleEndian, byte(len(msPayload.Parents))))
-	for _, parent := range msPayload.Parents {
-		if _, err := b.Write(parent[:]); err != nil {
-			panic(err)
-		}
-	}
-	if _, err := b.Write(msPayload.InclusionMerkleProof[:]); err != nil {
-		panic(err)
-	}
-
-	Must(binary.Write(&b, binary.LittleEndian, msPayload.NextPoWScore))
-	Must(binary.Write(&b, binary.LittleEndian, msPayload.NextPoWScoreMilestoneIndex))
-
-	Must(b.WriteByte(sigsCount))
-	for _, pubKey := range msPayload.PublicKeys {
-		if _, err := b.Write(pubKey[:]); err != nil {
-			panic(err)
-		}
-	}
-	Must(binary.Write(&b, binary.LittleEndian, uint32(0)))
-	Must(b.WriteByte(sigsCount))
-	for _, sig := range msPayload.Signatures {
-		if _, err := b.Write(sig[:]); err != nil {
-			panic(err)
-		}
-	}
-
-	return msPayload, b.Bytes()
-}
-
-// RandMilestoneSig returns a random milestone signature.
-func RandMilestoneSig() [iotago.MilestoneSignatureLength]byte {
-	var sig [iotago.MilestoneSignatureLength]byte
-	copy(sig[:], RandBytes(iotago.MilestoneSignatureLength))
-	return sig
+	return msPayload
 }
 
-// RandIndexation returns a random indexation payload.
-func RandIndexation(dataLength ...int) (*iotago.Indexation, []byte) {
-	const index = "寿司を作って"
-
+// RandTaggedData returns a random tagged data payload.
+func RandTaggedData(tag []byte, dataLength ...int) *iotago.TaggedData {
 	var data []byte
 	switch {
 	case len(dataLength) > 0:
```

Comment:
Should we add the possibility to test against data with zero length?

Reply:
If dataLength[0] is zero / `RandTaggedData([]byte("bla"), 0)`, then this will return a `TaggedData` with a zero data field.